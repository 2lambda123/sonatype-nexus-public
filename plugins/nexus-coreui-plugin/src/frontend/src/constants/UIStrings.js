/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import { UIStrings } from 'nexus-ui-plugin';
import React from 'react';

export default {
  ...UIStrings,

  ANONYMOUS_SETTINGS: {
    MENU: {
      text: 'Anonymous Access',
      description: 'Configure anonymous access to server contents'
    },
    ENABLED_CHECKBOX_LABEL: 'Access:',
    ENABLED_CHECKBOX_DESCRIPTION: 'Allow anonymous users to access the server',
    USERNAME_TEXTFIELD_LABEL: 'Username:',
    REALM_SELECT_LABEL: 'Realm:',
    MESSAGES: {
      LOAD_ERROR: 'An error occurred while loading Anonymous settings, see console for more details',
      SAVE_SUCCESS: 'Anonymous security settings updated',
      SAVE_ERROR: 'An error occurred while updating Anonymous settings, see console for more details'
    }
  },

  SYSTEM_INFORMATION: {
    MENU: {
      text: 'System Information'
    },
    ACTIONS: {
      download: 'Download as JSON'
    }
  },

  SUPPORT_REQUEST: {
    MENU: {
      text: 'Support Request',
      description: 'Submit a support request to Sonatype'
    },
    ACTIONS: {
      submitRequest: 'Submit Request'
    },
    DESCRIPTION: `Please include a complete description of your problem and steps to allow us to reproduce the problem (if available).`,
    ATTACH_SUPPORT_ZIP: `Attaching a support ZIP to your request will help our engineers give you a faster response.`
  },

  USER_ACCOUNT: {
    MENU: {
      text: 'Account',
      description: 'Manage your account',
    },
    ACTIONS: {
      changePassword: 'Change password',
      discardChangePassword: 'Discard password change',
    },
    MESSAGES: {
      LOAD_ERROR: 'An error occurred while loading User Account, see console for more details',
      UPDATE_SUCCESS: 'User account settings updated',
      UPDATE_ERROR: 'An error occurred while updating user account settings',
      PASSWORD_NO_MATCH_ERROR: 'Passwords do not match',
      PASSWORD_MUST_DIFFER_ERROR: 'New password must be different',
    },
    ID_FIELD_LABEL: 'Username',
    FIRST_FIELD_LABEL: 'First Name',
    LAST_FIELD_LABEL: 'Last Name',
    EMAIL_FIELD_LABEL: 'Email',
    PASSWORD_CURRENT_FIELD_LABEL: 'Current Password',
    PASSWORD_NEW_FIELD_LABEL: 'New Password',
    PASSWORD_NEW_CONFIRM_FIELD_LABEL: 'New Password (Confirm)',
  },
};
