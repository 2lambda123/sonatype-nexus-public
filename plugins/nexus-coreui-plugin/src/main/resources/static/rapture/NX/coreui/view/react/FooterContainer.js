/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Open Source Version is distributed with Sencha Ext JS pursuant to a FLOSS Exception agreed upon
 * between Sonatype, Inc. and Sencha Inc. Sencha Ext JS is licensed under GPL v3 and cannot be redistributed as part of a
 * closed source work.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
/*global Ext, NX, REACT_COMPONENTS, react, ReactDOM */

/**
 * React Footer Container
 *
 * @since 3.21
 */
Ext.define('NX.coreui.view.react.FooterContainer', {
  extend: 'Ext.Component',
  requires: [
    'NX.State',
    'NX.Permissions'
  ],

  alias: 'widget.nx-coreui-react-footer-container',

  reactView: undefined,

  reactViewProps: null,

  id: 'nxrm-react-footer-container',
  cls: 'nxrm-react-footer-container',

  listeners: {
    afterrender: 'initReactView',
    beforedestroy: 'destroyReactView'
  },

  initReactView: function() {
    const reactElement = react.createElement(this.reactView, {
      onClose: function() {
        this.dismissAlert()
      }.bind(this)
    }, null);
    ReactDOM.render(reactElement, this.getEl().dom);
  },

  destroyReactView: function() {
    if (this.reactView) {
      ReactDOM.unmountComponentAtNode(this.getEl().dom);
    }
  },

  refresh: function() {
    this.destroyReactView();
    this.initReactView();
    this.maybeSetVisible();
  },

  maybeSetVisible: function() {

    const isCircuitBreakerEnabled = NX.State.getValue('nexus.circuitb.enabled');
    const contentUsageEvaluationResults = NX.State.getValue('contentUsageEvaluationResult');
    const isHa = NX.State.getValue('nexus.datastore.clustered.enabled');
    const isProEdition = NX.State.getEdition() === 'PRO';
    const canViewMetrics = NX.Permissions.check('nexus:metrics:read');
    const softLimitCards = contentUsageEvaluationResults.filter(function(m) {
      return m.limitLevel === 'SOFT_LIMIT';
    });

    this.setVisible(!isProEdition && isCircuitBreakerEnabled && !isHa && softLimitCards.length > 0 && canViewMetrics);
    this.updateLayout();
  },

  dismissAlert: function(){
    this.getEl().child('div.nx-alert--warning').setDisplayed(false);
    this.updateLayout();
  },
});
