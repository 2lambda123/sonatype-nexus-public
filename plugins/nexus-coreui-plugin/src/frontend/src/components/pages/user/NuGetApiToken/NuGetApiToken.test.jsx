/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import React from 'react';
import Axios from 'axios';
import {act} from 'react-dom/test-utils';
import {fireEvent, render, wait} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import NuGetApiToken from './NuGetApiToken';
import UIStrings from '../../../../constants/UIStrings';

jest.mock('nexus-ui-plugin', () => {
  return {
    ...jest.requireActual('nexus-ui-plugin'),
    ExtJS: {
      requestAuthenticationToken: jest.fn(() => Promise.resolve({data: 'fakeToken'})),
      urlOf: jest.fn(() => 'http://localhost:4242/repository/fakeUrl'),
    }
  };
});


jest.mock('axios', () => {
  return {
    ...jest.requireActual('axios'),
    get: jest.fn((url) => {
      if (url === '/service/rest/internal/nugetApiKey?authToken=ZmFrZVRva2VuCg==') {
        return Promise.resolve({data: {data: {apiKey: 'testApiKey'}}});
      }
    }),
    delete: jest.fn((url) => {
      if (url === '/service/rest/internal/nugetApiKey?authToken=ZmFrZVRva2VuCg==') {
        return Promise.resolve({data: {data: {apiKey: 'newTestApiKey'}}});
      }
  }),
  };
});

describe('NuGetApiToken', () => {
  const renderView = async (view) => {
    let selectors = {};
    await act(async () => {
      let {container, getByText, queryByText} = render(view);
      selectors = {
        container,
        accessButton: () => getByText(UIStrings.NUGET_API_KEY.ACCESS.BUTTON),
        resetButton: () => getByText(UIStrings.NUGET_API_KEY.RESET.BUTTON),
        nugetKey: () => queryByText('testApiKey'),
        newNugetKey: () => queryByText('newTestApiKey')
      };
    });
    return selectors;
  };

  it('renders correctly', async () => {
    let { container, accessButton, resetButton, nugetKey } =
        await renderView(<NuGetApiToken/>);

    expect(container).toMatchSnapshot('baseline');
    await wait(() =>  expect(nugetKey()).not.toBeInTheDocument());

    await act(async () => fireEvent.click(accessButton()));

    await wait(() =>  expect(nugetKey()).toBeInTheDocument());

    expect(container).toMatchSnapshot('nugetKeyPresent');
  });

  it('uses the get call when the access button is pressed',  async () => {
    let {_, accessButton, resetButton, nugetKey  } =
        await renderView(<NuGetApiToken/>);

    await act(async () => fireEvent.click(accessButton()));

    await wait(() =>  expect(nugetKey()).toBeInTheDocument());

    expect(Axios.get).toHaveBeenCalledTimes(1);
    expect(Axios.get).toHaveBeenCalledWith(
        `/service/rest/internal/nugetApiKey?authToken=ZmFrZVRva2VuCg==`
    );
  });

  it('uses the delete call when the reset button is pressed',  async () => {
    let {_, accessButton, resetButton, nugetKey  } =
        await renderView(<NuGetApiToken/>);

    await act(async () => fireEvent.click(resetButton()));

    expect(Axios.delete).toHaveBeenCalledTimes(1);
    expect(Axios.delete).toHaveBeenCalledWith(
        `/service/rest/internal/nugetApiKey?authToken=ZmFrZVRva2VuCg==`
    );
  });
});
