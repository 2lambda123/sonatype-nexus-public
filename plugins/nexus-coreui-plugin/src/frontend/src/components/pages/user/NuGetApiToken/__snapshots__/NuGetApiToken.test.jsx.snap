// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NuGetApiToken renders correctly: baseline 1`] = `
<div>
  <div
    class="nxrm-content-body"
  >
    <div
      class="nxrm-settings-section"
    >
      <p>
         
        A new API Key will be created the first time it is accessed. Resetting your API Key will invalidate the current key.
         
      </p>
      <div
        class="nxrm-settings-section-footer"
      >
        <button
          class="nx-btn nx-btn--primary"
          type="input"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-lock fa-w-14 nx-icon"
            data-icon="lock"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
              fill="currentColor"
            />
          </svg>
           
          Access API Key
        </button>
        <button
          class="nx-btn"
          type="input"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-lock fa-w-14 nx-icon"
            data-icon="lock"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
              fill="currentColor"
            />
          </svg>
           
          Reset API Key
        </button>
      </div>
    </div>
  </div>
</div>
`;
